### DISCUSSION TITLE
Docksal install fails on Ubuntu 22.04 since docker ce 20.10.12.* no longer in repo

### DISCUSSION THREAD

**Discussion by purcelltastic** [docksal/docksal#1703] [Q&A]
Docker install fails from the script with the following error:

`INFO: apt-cache madison 'docker-ce' | grep '20.10.12.*-0~ubuntu' | head -1 | awk '{$1=$1};1' | cut -d' ' -f 3

ERROR: '20.10.12' not found amongst apt-cache madison results`

`apt-cache madison 'docker-ce'                             
 docker-ce | 5:20.10.17~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.16~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.15~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.14~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.13~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages`

Is there a way I can have the install script use a different version of docker ce?

**Comment by EJTH** [comment]
Same issue for me.

**Comment by ITedInnovator** [comment]
I experienced the same issue on Pop OS 22.04 which is Ubuntu 22.04. In the end I installed a stable version of docker-ce:
To find a stable version to install:

```
apt-cache policy docker-ce

apt-cache madison docker-ce

sudo apt-get install docker-ce=<VERSION_STRING> docker-ce-cli=<VERSION_STRING> containerd.io
```

It had started apache2 so I had to systemctl stop apache2 so port 80 was free after an address in use :80 error.

Now I don't know why Docksal claim to install Docker and warn when you have installed it yourself when the script doesn't work with Ubuntu 22.04. Is this only Ubuntu 22.04 LTS and if so is that a bug with Docksal or Ubuntu?
